tssc/step_implementers/unit_test/junit.py

```diff
@@ -107,8 +118,74 @@ def _run_step(self, runtime_step_config):
         dict
             Results of running this step.
         """
-        results = {}
+        pom_file = runtime_step_config['pom-file']
+
+        if not os.path.exists(pom_file):
+            raise ValueError('Given pom file does not exist: ' + pom_file)
+
+        maven_surefire_plugin = self.find_reference_in_pom(pom_file, 'maven-surefire-plugin')
+        if maven_surefire_plugin is None:
+            raise ValueError('Unit test dependency "maven-surefire-plugin" missing from POM.')
+
+        reports_dir = self.find_reference_in_pom(pom_file, 'reportsDirectory')
+        if reports_dir is not None:
+            default_test_results_dir = reports_dir
+        else:
+            default_test_results_dir = os.path.join(
+                os.path.dirname(os.path.abspath(pom_file)),
+                'target/surefire-reports')
+
+        test_results_output_path = self._StepImplementer__results_dir_path + '/unit-test/junit' # pylint: disable=no-member
+
+        try:
+            sh.mvn(  # pylint: disable=no-member
```

Comment:
@HunterGerlach shoulnd't this be wrapped in a `print` like the other places that @ppremru refactored?

Reply:
I took the print out and added _out=sys.stdout.

Reply:
oh. i thought we had to do both. I thought i saw some code somewhere that was doing both.

Reply:
I did not come across that (I just looked globally through all instances again). Personally, I prefer this method.

Reply:
ignore me. @ppremru streightned me out.